sandbox/apparmor/legacy.go

```diff
@@ -0,0 +1,85 @@
+package apparmor
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// Legacy represents apparmor user-space version up until 1.12
+//
+// Legacy apparmor uses the following structure:
+//
+//   /etc/apparmor.d/* (just files): source profiles.
+//   /etc/apparmor.d/tunables/*: definitions included by source profiles.
+//   /etc/apparmor.d/abstractions/*: definitions included by source profiles.
+//   /etc/apparmor.d/cache/*: binary profiles.
+//
+// In addition some locations are only used by snapd:
+//
+//   /var/lib/snapd/apparmor/profiles: source profiles (snapd specific)
+//   /var/cache/apparmor/*: binary profiles (snapd specific)
+//
+// Source profiles are typically stored as files in /etc/apparmor.d/.
+// The file name is the absolute path of the executable with forward slashes
+// replaced with dots. For example the file /etc/apparmor.d/usr.bin.man
+// contains a profile that attaches to /usr/bin/man. In addition the same file
+// defines additional profiles man_groff and man_filter, that various
+// executables started from /usr/bin/man assume.
+//
+// Source profiles routinely include definitions from
+// /etc/apparmor.d/abstractions as well as tunables that can be adjusted by
+// local administrator, from /etc/apparmor.d/tunables.
+//
+// Once compiled, binary profiles are typically cached as files in
+// /etc/apparmor.d/cache. The cache is based on mtime of the source profile and
+// of the cached profile. The cache is unaware of the kernel feature set.  This
```

Comment:
That's not true. /etc/apparmor.d/cache/.features is present prior to 2.13 and was used as part of the parser's decision to invalidate the cache.